wikipediapreview-wordpress.php

```diff
@@ -4,3 +4,27 @@
  * Description: Add Wikipedia Preview to your articles
  * Version: 0.1.0
  */
+
+function wikipediapreview_enqueue_scripts() {
+    wp_enqueue_script(
+        'wikipedia-preview', 
+        plugin_dir_url( __FILE__ ) . 'assets/js/wikipedia-preview.production.js', [], false, true 
+    );
+
+    wp_enqueue_style(
+        'wmf-wp-format-css',
+        plugin_dir_url( __FILE__ ) . 'assets/css/wikipedia-preview.css', [], false, true 
+    );
+
+    wp_enqueue_script( 'wikipedia-preview-init', plugin_dir_url( __FILE__ ) . 'init.js', [], false, true );	
+}
+
+// record the option of detect links feature enabled in this version,
+// detect links feature may disable by default in the next version
+function record_options() {
+    $options = array( 'auto_detects' => true );
+    add_option( 'wikipedia_preview_options', $options );
```

Comment:
Is there a way to set that option only once, maybe during plugin install or activation? And clean it up if the plugin gets removed.

Reply:
https://developer.wordpress.org/plugins/plugin-basics/activation-deactivation-hooks/

Reply:
thanks!

so 
activation : add option
deactivation : remove option
[uninstall ](https://developer.wordpress.org/reference/functions/register_uninstall_hook/): remove option

Reply:
oh wait, before uninstall, you need to deactivate the plugin first.

but is that possible any case to skip deactivation and uninstall it?

Reply:
I don't see a way to do it in the UI

Reply:
ok, it seems not possible, https://developer.wordpress.org/plugins/plugin-basics/uninstall-methods/, will remove it.

unless, the admin remove the plugin through the ftp manually, there is no way to detect it and will be very bad to the wordpress